views/update_project.erb

```diff
@@ -26,6 +26,10 @@
       <label for="webhookURL">Webhook URL</label>
       <input type="text" name="webhook_url" class="form-control"  id="webhookURL" value="<%= @project.webhook_url %>">
     </div>
+    <div class="form-group">
+      <label for="inputLineNotifyAccessToken">LINE Notify Access Token</label>
+      <input type="password" name="line_notify_access_token" class="form-control"  id="inputLineNotifyAccessToken" value="<% if @project.line_notify_access_token %>***<% end %>">
```

Comment:
Thank you for your comments!

Actually, I understand that current implementation is bad because for example, if user try to change a hipchat room without editing LINE Notify Access Token, user involuntarily sets the access token to '***'.

So I think that it's better to divide '/project/*/update' screen.

* show_project
![](https://gyazo.com/6cea4b836e28ab6c5659733408e0921c.png)

* update_project
![](https://gyazo.com/ab4aa8da777cfde0874a7711ab4a3916.png)

* update_line_notify_access_token
![](https://gyazo.com/21193bf001b1cba318be4a7a488d23ab.png)

Even though this implementation, if user push the Update button without editing token, '***' is set.
But I hope no such users...

What do you think?